storage/client/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -1,28 +1,276 @@
+import com.azure.storage.queue.models.PeekedMessage;
+import com.azure.storage.queue.models.QueueGetPropertiesHeaders;
+import com.azure.storage.queue.models.QueueMessage;
+import com.azure.storage.queue.models.QueueProperties;
+import com.azure.storage.queue.models.QueuesGetPropertiesResponse;
+import com.azure.storage.queue.models.SignedIdentifier;
+import com.azure.storage.queue.models.UpdatedMessage;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
 import java.net.URL;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
 
-public class QueueAsyncClient extends ServiceClient {
-    private final String endpoint;
-    private final AzureQueueStorageImpl generateClient;
-    private final String apiVersion;
+/**
+ * Queue async client
+ */
+public final class QueueAsyncClient {
+    private final AzureQueueStorageImpl client;
+    private final String queueName;
 
-    private QueueAsyncClient(URL endpoint, HttpPipeline httpPipeline) {
-        super(httpPipeline);
-        this.endpoint = endpoint.toString();
-        this.generateClient = new AzureQueueStorageImpl(httpPipeline).withUrl(this.endpoint);
-        this.apiVersion = this.generateClient.version();
+    /**
+     * Constructor used by QueueServiceAsyncClient
+     * @param client Client that interacts with the service interfaces
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(AzureQueueStorageImpl client, String queueName) {
+        this.queueName = queueName;
+        this.client = new AzureQueueStorageImpl(client.httpPipeline())
+            .withUrl(client.url())
+            .withVersion(client.version());
+    }
+
+    /**
+     * Constructor used by the builder
+     * @param endpoint URL of the queue
+     * @param httpPipeline Http pipeline
+     * @param queueName Name of the queue
+     */
+    QueueAsyncClient(URL endpoint, HttpPipeline httpPipeline, String queueName) {
+        this.queueName = queueName;
+        this.client = new AzureQueueStorageImpl(httpPipeline)
+            .withUrl(endpoint.toString());
     }
 
     /**
-     * Creates a builder that can configure options for the SecretAsyncClient before creating an instance of it.
-     * @return A new builder to create a SecretAsyncClient from.
+     * @return a new client builder instance
      */
     public static QueueAsyncClientBuilder builder() {
         return new QueueAsyncClientBuilder();
     }
+
+    /**
+     * @return the URL of the queue
+     */
+    public String url() {
+        return client.url();
+    }
+
+    /**
+     * Creates a queue with no metadata
+     * @return an empty response
+     */
+    public Mono<VoidResponse> create() {
+        return create(null);
+    }
+
+    /**
+     * Creates a queue with metadata
+     * @param metadata Metadata to set on the queue
+     * @return an empty response
+     */
+    public Mono<VoidResponse> create(Map<String, String> metadata) {
```

Comment:
Why VoidResponse instead of Void?

Reply:
VoidResponse is used to comply with the requirements that the clients need to have the HTTP headers and other data available in the response.